components/shared/MemoPopup.tsx

```diff
@@ -0,0 +1,113 @@
+'use client';
+
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+import { ActionButton } from '../common/ActionButton';
+import { SectionTitle } from '@/components/shared/SectionTitle';
+import { DatePickerSection } from './DatePickerSection';
+import { CalendarIcon, StarIcon, FileTextIcon, PenLine, X } from 'lucide-react';
+import Card from '@/components/common/Card';
+
+const memoSchema = z.object({
+  title: z.string().optional(),
+  content: z.string().min(1, '내용은 필수입니다.'),
+  startDate: z.date({ required_error: '시작일을 선택해주세요.' }),
+  endDate: z.date({ required_error: '종료일을 선택해주세요.' }),
```

Comment:
추가로 아래 처럼 하면 시작일이 종료일보다 뒤에 올 경우도 잡아낼 수 있을 거 같아요
```suggestion
  endDate: z.date({ required_error: '종료일을 선택해주세요.' }),
  })
  .refine((data) => data.startDate < data.endDate, {
    path: ['endDate'],
    message: '종료일은 시작일 이후여야 합니다.',
  });
```

Reply:
수정 완료 했습니다!!